### PR TITLE
Generated design picker: Fetch recipes for generated designs via API 

### PR DESCRIPTION
**Pull request by taipeicoder**
#### Changes proposed in this Pull Request

This PR retrieves recipes for generated designs from the backend instead of a local file. This allows us to manage the list of recipes without needing to update the front-end.  

#### Testing instructions

* Make sure you can access the generated design picker screen by:
  * Having the necessary flags enabled.
  * Having a vertical selected from the vertical question screen.
* Head to `/setup/intent?siteSlug=${site_slug}`, and click on the "Start building" button.
* Expect the list of generated designs to match the list from the `block-recipes` [API endpoint](https://public-api.wordpress.com/wpcom/v2/block-recipes).

Related to #62237

### CODE REVIEW

File: client/landing/stepper/stores.ts [hunk 1]
```diff
@@ -22,3 +23,5 @@ export const USER_STORE = User.register( {
 } );
 
 export const AUTOMATED_ELIGIBILITY_STORE = AutomatedTransferEligibility.register();
+
+export const BLOCK_RECIPES_STORE = BlockRecipes.register();
```
Comment (fushar) [inline client/landing/stepper/stores.ts]: Hmm, since the recipes are used only in the design picker, I wonder if it is much simpler to just fetch them using a react-query hook (`useGeneratedDesigns()`), similar to the existing logic in `useThemeDesignsQuery()`?

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx [hunk 2]
```diff
@@ -90,10 +90,11 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 		[ staticDesigns ]
 	);
 
-	const generatedDesigns = useGeneratedDesigns();
+	const { data: generatedDesigns, isLoading: isLoadingGeneratedDesigns } =
+		useGeneratedDesignsQuery();
 	const shuffledGeneratedDesigns = useMemo(
-		() => shuffle( generatedDesigns ),
-		[ generatedDesigns ]
+		() => ( ! isLoadingGeneratedDesigns ? shuffle( generatedDesigns ) : [] ),
```
Comment (fushar) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx]: I think we can revert these changes by simply make `[]` as the default value of `generatedDesigns` above:

```
const { data: generatedDesigns = [], ...
```

(because `shuffle([])` is still `[]`)

Comment (taipeicoder) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx]: Good point 👍

File: packages/design-picker/src/hooks/use-generated-designs-query.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+function apiBlockRecipeToDesign( recipe: BlockRecipe ): Design {
+	const { slug, title, stylesheet, pattern_ids } = recipe;
+
+	return {
+		slug,
+		title,
+		recipe: { stylesheet, patternIds: pattern_ids },
+		categories: [ { slug, name: title } ],
```
Comment (fushar) [inline packages/design-picker/src/hooks/use-generated-designs-query.ts]: Do we still need generated designs to have a category? Where is it being used?

Comment (taipeicoder) [reply packages/design-picker/src/hooks/use-generated-designs-query.ts]: Not used but same as `features`, `template`, and `theme` it's currently [not an optional property](https://github.com/Automattic/wp-calypso/blob/trunk/packages/design-picker/src/types.ts#L29).

Comment (fushar) [reply packages/design-picker/src/hooks/use-generated-designs-query.ts]: I see, I'd suggest set is as `[]` then to reduce confusion?

Comment (taipeicoder) [reply packages/design-picker/src/hooks/use-generated-designs-query.ts]: `slug` and `name` are also non-optional 🥲

Comment (fushar) [reply packages/design-picker/src/hooks/use-generated-designs-query.ts]: I mean

```
categories: [],
```

Comment (taipeicoder) [reply packages/design-picker/src/hooks/use-generated-designs-query.ts]: Ah, okay 👍

**Review by fushar — APPROVED** [review]
Tested & works 👍
